auth/src/main/scala/auth/repo/UserRepositoryImpl.scala

```diff
@@ -0,0 +1,58 @@
+package repo
+
+import model.User
+import zio.{ZIO, ZLayer}
+import zio.sql.ConnectionPool
+import zio.stream.ZStream
+
+final class UserRepositoryImpl(
+                                    pool: ConnectionPool
+                                  ) extends UserRepository
+  with PostgresTableDescription {
+
+  val driverLayer: ZLayer[Any, Nothing, SqlDriver] =
+    ZLayer
+      .make[SqlDriver](
+        SqlDriver.live,
+        ZLayer.succeed(pool)
+      )
+
+  def EncryptPassword(password: String): String = {
+    val crypt = java.security.MessageDigest.getInstance("SHA-1")
```

Comment:
https://crypto.stackexchange.com/questions/48289/how-secure-is-sha1-what-are-the-chances-of-a-real-exploit